modules/transactionpool/accept.go

```diff
@@ -192,21 +198,35 @@ func (tp *TransactionPool) addTransactionToPool(t consensus.Transaction, directi
 // AcceptTransaction takes a new transaction from the network and puts it in
 // the transaction pool after checking it for legality and consistency.
 func (tp *TransactionPool) AcceptTransaction(t consensus.Transaction) (err error) {
+	// Check that the transaction has not been seen before.
+	id := tp.mu.Lock()
+	txnHash := crypto.HashObject(t)
+	_, exists := tp.seenTransactions[txnHash]
+	if exists {
+		tp.mu.Unlock(id)
+		return ErrDuplicate
+	}
+	if len(tp.seenTransactions) > 1200 {
+		tp.seenTransactions = make(map[crypto.Hash]struct{})
+	}
```

Comment:
hmm. Seems less than ideal to just wipe the map entirely. Limiting the set to the 1200 most recent txns would be better, but more of a pain to implement. I guess we'll leave it for now.